### ISSUE TITLE
Error with --map-by core

### ISSUE THREAD

**Issue by afernandezody** [open-mpi/ompi#8049]
Hello OpenMPI Team,
I'm observing a similar behavior to that reported in #7867 on a CentOS 8 system with the error messages:
```
[eap-ac01-000004:663344] PMIX ERROR: NOT-FOUND in file dstore_base.c at line 2866
[eap-ac01-000004:663344] PMIX ERROR: NOT-FOUND in file server/pmix_server.c at line 3408
[eap-ac01-000004:663348] PMIX ERROR: OUT-OF-RESOURCE in file client/pmix_client.c at line 231
[eap-ac01-000004:663348] OPAL ERROR: Error in file pmix3x_client.c at line 112
*** An error occurred in MPI_Init
*** on a NULL communicator
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)
[eap-ac01-000004:663348] Local abort before MPI_INIT completed completed successfully, but am not able to aggregate error messages, and not able to guarantee that all other processes were killed!
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[22884,1],0]
  Exit code:    1
--------------------------------------------------------------------------
[eap-ac01-000004:663344] PMIX ERROR: ERROR in file gds_ds21_lock_pthread.c at line 99
[eap-ac01-000004:663344] PMIX ERROR: ERROR in file gds_ds21_lock_pthread.c at line 99

```
When I turn off the error (`export PMIX_MCA_gds=hash`) or install v4.1.x, the system doesn't complain, but it just hangs in there doing nothing. Thanks.

**Comment by hppritcha** [comment]
@afernandezody which version of Open MPI are you using?

**Comment by afernandezody** [comment]
The error posted above was using v4.0.4. I also tried v4.1.x (master) but this version doesn't generate any error (it simply idles).

**Comment by hppritcha** [comment]
This issue should be fixed in 4.0.5.  it was recently released.  Please give it a try.

**Comment by afernandezody** [comment]
It does work with 4.0.5 (I guess that some components of 4.1.x might be a bit behind 4.0.5 but will be updated at a later time). Thanks.

**Comment by jsquyres** [comment]
@afernandezody Note that `master` is actually what will become 5.0.0.  There is a separate v4.1.x branch - I'd be curious if your problem is fixed on the 4.1.x branch as well (we're about to make another rc for 4.1.0 -- I don't know if the fix is included in 4.1.0rc1 or not).

**Comment by afernandezody** [comment]
Hi @**jsquyres**. I apologize for the delay as I was testing a bunch of things. After cleaning one of the systems and reinstalling v4.1.x, the option --map-by core worked w/o any problems (my guess is that I had mixed some of the library paths or some other silly error). P.S. I was aware that v4.1.x will eventually become v5. Thanks.
